### PR TITLE
Moved transform_devicons & get_devicons from make_entry.lua to utils.lua

### PR DESCRIPTION
**Pull request by ndavidq0**
...

@Conni2461

### CODE REVIEW

File: lua/telescope/utils.lua [hunk 1]
```diff
@@ -295,4 +298,46 @@ utils.align_str = function(string, width, right_justify)
     or string..string.rep(" ", width - str_len)
 end
 
+if has_devicons then
+  if not devicons.has_loaded() then
+   devicons.setup()
+  end
+
+  utils.transform_devicons = function(filename, display, disable_devicons)
```
Comment (Conni2461) [inline lua/telescope/utils.lua:306]: I would do this differently and i think that is the issues:
```lua
utils.transform_devicons = (function()
  if has_devicons then
    -- load devicons
    return function(filename, display, disable_devicons)
      -- the content of transform_devicons
    end
  else
    return function(_, display, _)
      return display
    end
  end
end)()
```

And then the same for `get_devicons`

The function will than run the first time and will initialize `transform_devicons` and `get_devicons`

Does this make sense for you?

File: lua/telescope/utils.lua [hunk 2]
```diff
@@ -295,4 +297,50 @@ utils.align_str = function(string, width, right_justify)
...
+      local icon, icon_highlight = devicons.get_icon(filename, string.match(filename, '%a+$'), { default = true })
+      if conf.color_devicons then
+        return icon, icon_highlight
+      else
+        return icon
+      end
+    end
+  else
+    return function(_, display, _)
+      return display
```
Comment (Conni2461) [inline lua/telescope/utils.lua:341]: This function was only `(_, _) return ''` it doesn't have a display. It just returns the icon and highlight

Comment (ndavidq0) [reply lua/telescope/utils.lua:341]: Oh didn't notice it

**Review by Conni2461 — APPROVED** [review]
LGTM

Thanks :)

**Pull request merged**